src/exchange_translator.h

```diff
@@ -80,22 +83,27 @@ class ExchangeTranslator {
         ex_ctx_->trader_prefs.at(req->requester())[req][bid];
     if (pref < 0) {
       CLOG(LEV_DEBUG1) << "Removing arc because of negative preference.";
-    } else {
-      // get translated arc
-      Arc a = TranslateArc(xlation_ctx_, bid);
-
-      a.unode()->prefs[a] = pref;  // request node is a.unode()
-      int n_prefs = a.unode()->prefs.size();
-
-      CLOG(LEV_DEBUG5) << "Updating preference for one of "
-                       << req->requester()->manager()->prototype()
-                       << "'s trade nodes:";
-      CLOG(LEV_DEBUG5) << "   preference: " << a.unode()->prefs[a];
-
-      graph->AddArc(a);
+      return;
+    } else if (pref == 0) {
+      std::stringstream ss;
+      ss << "0-valued preferences have been deprecated. "
+         << "Please make preference value positive.";
```

Comment:
Add to this message "This error will go away in the next release".  And make an issue to remove the throw for the following release.